src/XamlX.IL.Cecil/CecilTypeSystem.cs

```diff
@@ -41,7 +42,7 @@ CecilAssembly ResolveWrapped(AssemblyNameReference name)
             foreach (var asm in _asms)
                 if (asm.Assembly.Name.Name == name.Name)
                     return _assemblyCache[name.FullName] = asm;
-            throw new AssemblyResolutionException(name);
+            return throwOnNotFound ? throw new AssemblyResolutionException(name) : null;
```

Comment:
Current method returns null for CoerceReference.
While building Avalonia assemblies there was no "unknown" references, so we can remove "throwOnNotFound" and just throw always.
Although, I am not so sure about every possible type used in XAML.